### DISCUSSION TITLE
Can I make the library loaded via envvar `LD_PRELOAD` also be executed within `firejail`?

### DISCUSSION THREAD

**Discussion by kuanyui** [netblue30/firejail#5552] [Q&A] [answered]
I've look around the issue trackers and found these:

- https://github.com/netblue30/firejail/issues/460#issuecomment-213695062
- https://github.com/netblue30/firejail/discussions/5512

But I still don't sure the differences between:

1. `firejail sh -c "LD_PRELOAD='foo.so bar.so' main_app"`
2. `firejail --env=LD_PRELOAD='foo.so bar.so' main_app`

The scenario I encountered is, I want to use `--net=none` to disable all Internet access of `main_app`, but to run this `main_app`, I need to load some (maybe closed-source) shared libraries like `foo.so` and `bar.so` via `LD_PRELOAD`. 

The question is I don't sure if the above two ways can also disable Internet access of shared libraries of `foo.so` and `bar.so` ? 

I read `man firejail` and found this:

```
       --env=name=value
              Set environment variable in the new sandbox.

              Example:
              $ firejail --env=LD_LIBRARY_PATH=/opt/test/lib
```

That is to say... `firejail --env=LD_PRELOAD='foo.so'` will make `foo.so` also loaded within `firejail` right?

> Sorry I don't sure if I'm asking a dumb question because I'm not familiar with Linux programming.

Thanks.

**Comment by rusty-snake** [comment, marked as answer]
1. and 2. will make your secret program to load the library (as long as the file is accessiable inside the sandbox, the program is a dynamic linked ELF, ... the obvious conditions). However 1. requires a shell inside the sandbox and uses the default profile while 2. uses the profile for your secret program if any and does not require a shell.
